hal/common/retarget.cpp

```diff
@@ -624,7 +632,7 @@ extern "C" void software_init_hook(void)
         mbed_die();
     }
 #endif/* FEATURE_UVISOR */
-
+    mbed_sdk_init();
```

Comment:
This is before the RTOS has been initialized which might case problems on other devices. I'll trigger testing to check this.